### ISSUE TITLE
Add negative index support

### ISSUE THREAD

**Issue by Kikobeats** [Kikobeats/array-future#5]
https://github.com/sindresorhus/negative-array

Related:

- https://github.com/sindresorhus/cycled

**Comment by lukewestby** [comment]
Is this possible to implement efficiently?

**Comment by lukewestby** [comment]
Without an ES6 Proxy, that is

**Comment by Kikobeats** [comment]
I do not think so. The reference is for inspiration, but will be difficult bring this feature in the library.

Add others polyfills is not a good idea, because increases considerably the size of the library.

In addition, ES6 Proxy is poor supported, and the current most famous polyfill need special node/browser flags. I would like have this in the library, but too much friction for the final user that is the thing that we try to avoid.

**Comment by mk-pmb** [comment]
I think the next best thing to negative indexes were if the .last() method could accept (n) and return the [length - n]th element.

**Comment by Kikobeats** [comment]
@mk-pmb can you provide example of your idea?

**Comment by mk-pmb** [comment]
``` javascript
function guessScriptFileBeforeCurrent() {
  return Array.from(document.getElementsByTagName('script')).last(2).src;
}
```

**Comment by Kikobeats** [comment]
I think nowadays make sense ship this feature now that Node.js supports Proxy :)

Added https://github.com/sindresorhus/cycled as inspiration, should be done very similar.

Any PR? 🎉
